src/main/java/com/hida/Minter.java

```diff
@@ -0,0 +1,475 @@
+                } while (ID_LIST.size() < this.AMOUNT);
+            return convertListToJson();
+        } else {
+            return "cannot make any more ids given parameters";
+        }
+
+    }
+
+    /**
+     * Calculates the number of possible permutations in a given char mapping
+     * @return - number of permutations
+     */
+    private long calculateCharMapPermutations() {
+        long numPermutations = 1;
+        for (int i = 0; i < CHAR_MAP.length(); i++) {
+            /* when it isn't auto, we will not check the database as that
+             will not produce an accurate reseult. What will be returned 
+             instead is the comparison between the maximum number of 
+             permutations and the requested amount.
+             */
+
+            String charToken = BASE_MAP.get(CHAR_MAP.charAt(i));
+            System.out.println("charToken = " + charToken);
+            int base = tokenMaps.get(charToken).length();
+            System.out.println("base = " + base);
+            numPermutations *= base;
+            System.out.println("numPermutations = " + numPermutations);
+        }
+        return numPermutations;
+    }
+    
+    
+    /**
+     * Generates random ids based on char mappings. The ids are stored in
+     * ID_LIST
+     * @param numIdsToMake - number of ids to make     
+     */
+    private void randomAlgorithm(int numIdsToMake){          
+        String tempId, charToken;
+        String charRange;
+        int randomNum;
+                
+        while (ID_LIST.size() < numIdsToMake) {
+            tempId = PREFIX; 
+            for (int j = 0; j < CHAR_MAP.length(); j++) {
+                charToken = BASE_MAP.get(CHAR_MAP.charAt(j));
+                charRange = tokenMaps.get(charToken);
+                randomNum = (int)(Math.random() * charRange.length());
+                
+                tempId += charRange.charAt(randomNum);
+                                                
+            }
+            ID_LIST.add(tempId);
+        }
+                
+    }
+
+    /**     
+     * In the for loop, k represents last index of char mapping. This for 
+     * loop will incrementally increase the last digit until it reaches its
+     * max value, designated by CHAR_MAP and BASE_MAP. If there's 
+     * overflow, k is incremented so that the next digit can also be
+     * incremented. This process continues until there is no overflow. The
+     * last sequence is returned to the caller. 
+     * @param numIdsToMake 
+     * @param baseMap
+     */
+    private int[] sequentialAlgorithm(int numIdsToMake, int[] baseMap) {
+        
+        StringBuilder buffer = new StringBuilder();
+        
+
+        // define the last position of the map
+        int lastPos = CHAR_MAP.length() - 1;
+
+        buffer.append(PREFIX);
+
+        // prepare storing the number of ids that will be created
+        int idsCreated = 0;
+        
+        for (int k = 0; idsCreated < numIdsToMake;) {

```

Comment:
Looking at this is very confusing.  It's usually good practice to define iteration variables only within the `for` loop.  For  something like this, I'd sugest defining `k` outside of the loop, and turning this into a `while` loop:

```java
int k = 0;
while (idsCreated < numIdsToMake) {
    ...
}
```